src/SFML/Network/IpAddress.cpp

```diff
@@ -194,23 +160,72 @@ IpAddress IpAddress::getPublicAddress(Time timeout)
 
 
 ////////////////////////////////////////////////////////////
+void IpAddress::resolve(const std::string& address)
+{
+    m_address = 0;
+    m_valid = false;
+
+    if (address == "255.255.255.255")
+    {
+        // The broadcast address needs to be handled explicitly,
+        // because it is also the value returned by inet_addr on error
+        m_address = INADDR_BROADCAST;
+        m_valid = true;
+    }
+    else if (address == "0.0.0.0")
+    {
+        m_address = INADDR_ANY;
+        m_valid = true;
+    }
+    else
+    {
+        // Try to convert the address as a byte representation ("xxx.xxx.xxx.xxx")
+        sf::Uint32 ip = inet_addr(address.c_str());
+        if (ip != INADDR_NONE)
+        {
+            m_address = ip;
+            m_valid = true;
+        }
+        else
+        {
+            // Not a valid address, try to convert it as a host name
+            addrinfo hints;
+            std::memset(&hints, 0, sizeof(hints));
+            hints.ai_family = AF_INET;
+            addrinfo* result = NULL;
+            if (getaddrinfo(address.c_str(), NULL, &hints, &result) == 0)
+            {
+                if (result)
+                {
+                    ip = reinterpret_cast<sockaddr_in*>(result->ai_addr)->sin_addr.s_addr;
+                    freeaddrinfo(result);
+                    m_address = ip;
+                    m_valid = true;
+                }
+            }
+        }
+    }
+}
+
+
+////////////////////////////////////////////////////////////
 bool operator ==(const IpAddress& left, const IpAddress& right)
 {
-    return left.toInteger() == right.toInteger();
+    return !(left < right) && !(right < left);
 }
 
 
 ////////////////////////////////////////////////////////////
 bool operator !=(const IpAddress& left, const IpAddress& right)
 {
-    return !(left == right);
+    return (left < right) || (right < left);
 }
 
 
 ////////////////////////////////////////////////////////////
 bool operator <(const IpAddress& left, const IpAddress& right)
 {
-    return left.toInteger() < right.toInteger();
+    return (left.m_valid < right.m_valid) || (left.m_address < right.m_address);
```

Comment:
This violates [strict weak ordering](https://en.wikipedia.org/wiki/Weak_ordering#Strict_weak_orderings) and makes `sf::IpAddress` unusable in associative containers.

For example, consider the following objects:
```cpp
IpAddress a { m_valid = true, m_address = 0 };
IpAddress b { m_valid = false, m_address = 1 };
```
Both `a < b` and `b < a` are true.

You should always be overly cautious when using logical operators to implement relational operators. It's so easy to break behavior. A fool-proof and simple approach is to delegate the comparison to the standard library:
```cpp
// C++98
return std::make_pair(left.m_valid, left.m_address) < std::make_pair(right.m_valid, right.m_address);

// C++11
return std::make_tuple(/* all left attributes */) < std::make_pair(/* all right attributes */); // by value
return std::tie(/* all left attributes */) < std::tie(/* all right attributes */); // by reference
```